xcvm/architecture/transports/otp.md

```diff
@@ -0,0 +1,208 @@
+
+The following diagram describes the flow of data (and funds) when OTP is used to for ingress of programs.
+
+```mermaid
+sequenceDiagram
+    Third-party transport->>Third-party OTP-adapter: Transform messsage.
+    Third-party OTP-adapter->>Gateway: Transfer assets and pass program.
+    Gateway->>Router: Add bridging info and transfer funds.
+    Note over Gateway: Accept/deny program based on blacklist.
+    Router->>XCVM Interpreter: Instantiate vm and transfer funds.
+    Note over XCVM Interpreter: Accept/deny program based on gateway annotation.
+```
+
+Although the diagram simplifies the work that the OTP-adapter might need to perform, it can be relatively complicated depending on how the transport operates. We go into more concrete implementations [later](#possible-implementations) in this document. 
+
+## Registration of Routes
+
+For a transport to be possible be selected by the Gateway to transfer messages to different chains, a more involved registration process is required. 
+
+- A stake needs to be put down, which is used to reward disputers in case of bridge fraud.
+- Metadata on the bridge must be provided, including 
+    1. The `Account` of the administrator, used to deregister the bridge, and add routes.
+    2. Destinations reachable through this bridge.
+
+```
+RegistrationInfo ::= Account [Destination]
+Destination      ::= NetworkId [AssetId]
+```
+
+Immediately after registration, the bridge registration can be disputed for approximately 72 hours. The stake can be lost during this time, if governance determines that the registration is fraudulent. If a dispute is triggered, the registration can be delayed by up to 2 weeks.
+
+#### Disputed Registration
+
+The following diagram displays the sequence of a registration, including the dispute by a validator.
+
+```mermaid
+sequenceDiagram
+    Third-party Transport->>Gateway: Register bridge for egress.
+    Note over Gateway: Dispute window is opened.
+    Validator->>Gateway: Dispute the registration.
+    Note over Validator, Gateway: Validator puts down a stake as well.
+    Governance->>Gateway: Settle dispute.
+```
+
+Validators can be any entity, including a user triggering the dispute through our UI, as well as nodes automatically tracking registrations. For bridging teams, it is especially important to dispute registrations of your bridge with an incorrect administrator account.
+
+## Egress of messages
+
+Once a transport has been succesfully registered, it becomes accessible to the `Gateway`, which might route traffic based on transport requirements from the interpreter instance dispatching the program. By default, all OTP transports are considered `Trusted`, meaning that no traffic is routed through them unless specifically opted-in by the interpreter instance. 
+
+Later we may add means through governance to upgrade transports to the `Trustless` status using governance. This will accomodate transports such as `XCMP`.
```

Comment:
```suggestion
Later we may add means to upgrade transports to the `Trustless` status using governance. This will accomodate transports such as `XCMP`.
```